src/models/propulsion/FGBrushLessDCMotor.cpp

```diff
@@ -0,0 +1,236 @@
+    CoilResistance = el->FindElementValueAsNumber("coilresistance");
+  if (el->FindElement("noloadcurrent"))
+    NoLoadCurrent = el->FindElementValueAsNumber("noloadcurrent");
+  if (el->FindElement("deceleration_factor"))
+    DecelerationFactor = el->FindElementValueAsNumber("deceleration_factor");
+
+  MaxCurrent = MaxVolts / CoilResistance + NoLoadCurrent;
+
+  PowerWatts = MaxCurrent * MaxVolts;
+
+  string base_property_name = CreateIndexedPropertyName("propulsion/engine",
+    EngineNumber);
+  exec->GetPropertyManager()->Tie(base_property_name + "/power-hp", &HP);
+
+  exec->GetPropertyManager()->Tie(base_property_name + "/current-a", &CurrentRequired);
+
+  Debug(0); // Call Debug() routine from constructor if needed
+}
+
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+FGBrushLessDCMotor::~FGBrushLessDCMotor()
+{
+  Debug(1); // Call Debug() routine from constructor if needed
+}
+
+
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+void FGBrushLessDCMotor::Calculate(void)
+{
+  RunPreFunctions();
+
+  if (Thruster->GetType() == FGThruster::ttPropeller) {
+    ((FGPropeller*)Thruster)->SetAdvance(in.PropAdvance[EngineNumber]);
+    ((FGPropeller*)Thruster)->SetFeather(in.PropFeather[EngineNumber]);
+  }
+
+  RPM = Thruster->GetRPM();
+  TorqueRequired = abs(((FGPropeller*)Thruster)->GetTorque()); //units [#*ft]
+
+  CurrentRequired = (TorqueRequired * VelocityConstant) / TorqueConstant;
+
+  //  total current required must include no load current i0
+  CurrentRequired = CurrentRequired + NoLoadCurrent;
+ 
+  V = MaxVolts * in.ThrottlePos[EngineNumber];
+  
+  //  Delta RPM = (input voltage - currentRequired * coil resistance) * velocity costant
+  DeltaRPM = round((V - CurrentRequired * CoilResistance) * VelocityConstant);
+
+  //  Torque is MaxTorque (stall torque) at 0 RPM and linearly go to 0 at max RPM (MaxVolts*VelocityCostant)
+  //  MaxTorque = MaxCurrent*torqueconstant/velocityconstant*(1-RPM/maxRPM)
+
+  MaxTorque = MaxCurrent / VelocityConstant * TorqueConstant * (1 - RPM / (MaxVolts* VelocityConstant));
+
+  TorqueAvailable = MaxTorque - TorqueRequired;
+  InertiaTorque = (((DeltaRPM/60)*(2.0 * M_PI))/(max(0.00001, in.TotalDeltaT))) * ((FGPropeller*)Thruster)->GetIxx();
+ 
+  //  compute acceleration and deceleration phases:
+  //  Acceleration is due to the max delta torque available and is limited to the inertial forces
+
+  if (DeltaRPM >= 0) {
+    TargetTorque = min(InertiaTorque, TorqueAvailable) + TorqueRequired;
+  } else {
+  //  Deceleration is due to braking force given by the ESC and set by parameter deceleration_time 
+    TargetTorque = TorqueRequired - min(abs(InertiaTorque)/(max(DecelerationFactor,0.01)*30),RPM*TorqueConstant/VelocityConstant/VelocityConstant/CoilResistance);
+  }
+
+  EnginePower = ((2 * M_PI) * max(RPM, 0.0001) * TargetTorque) / 60;   //units [#*ft/s]
+  HP = EnginePower / 550; // units[HP]
```

Comment:
engine power is in  pounds*ft/s since torque is in ft*pounds. So conversion factor is not 745.7. but 745/1.36.
I added the comments with [units] to make all this understandable......... 
I'd rather prefer to use only metric units...